Building and running issue related to PR #166

After PR #166. Some building and running error were found here. Could anyone try to repeat it?

Linking CXX executable bin/test_welldensitysegmented
[100%] Built target test_welldensitysegmented
In file included from /home/kaib/learning/Git/OPM/build/opm-autodiff/tests/test_transmissibilitymultipliers.cpp:27:0:
/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/GeoProps.hpp: In member function ‘void Opm::DerivedGeology::multiplyHalfIntersections_(const Grid&, Opm::EclipseStateConstPtr, const std::vector<double>&, Opm::DerivedGeology::Vector&, std::vector<double>&) [with Grid = UnstructuredGrid; Opm::EclipseStateConstPtr = std::shared_ptr<const Opm::EclipseState>; Opm::DerivedGeology::Vector = Eigen::Array<double, -1, 1>]’:
/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/GeoProps.hpp:215:47: error: expected primary-expression before ‘,’ token
                     OPM_THROW(std::logic_error, "Unhandled face direction: " << faceTag);
                                               ^
/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/GeoProps.hpp:215:81: error: invalid operands of types ‘const char [27]’ and ‘int’ to binary ‘operator<<’
                     OPM_THROW(std::logic_error, "Unhandled face direction: " << faceTag);
                                                                                 ^
/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/GeoProps.hpp:215:88: error: ‘OPM_THROW’ was not declared in this scope
                     OPM_THROW(std::logic_error, "Unhandled face direction: " << faceTag);
                                                                                        ^
make[2]: **\* [CMakeFiles/test_transmissibilitymultipliers.dir/tests/test_transmissibilitymultipliers.cpp.o] Error 1
make[1]: **\* [CMakeFiles/test_transmissibilitymultipliers.dir/all] Error 2
make[1]: **\* Waiting for unfinished jobs....
Linking CXX executable bin/sim_fibo_ad
[100%] Built target sim_fibo_ad

When Running SPE9

---------------    Simulation step number 2    ---------------
      Current time (days)     2
      Current stepsize (days) 2
      Total time (days)       900

Iteration         Residual
        0       0.705650991
=== RestartedGMResSolver
=== GMRes::restart
=== rate=0.446070214, T=0.18565, TIT=0.018565, IT=10
        1       0.144569303
Program threw an exception: [/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/NewtonIterationBlackoilCPR.cpp:243] Cannot do Schur complement with respect to non-diagonal block.
terminate called after throwing an instance of 'std::logic_error'
  what():  [/home/kaib/learning/Git/OPM/build/opm-autodiff/opm/autodiff/NewtonIterationBlackoilCPR.cpp:243] Cannot do Schur complement with respect to non-diagonal block.
Aborted (core dumped)

Something also wrong with SPE3, while not that serious.

--- a/examples/sim_fibo_ad.cpp
+++ b/examples/sim_fibo_ad.cpp
@@ -193,7 +193,7 @@ try
     // initialize variables
     simtimer.init(timeMap);
 
-    Opm::DerivedGeology geology(*grid->c_grid(), *new_props, eclipseState);
+    Opm::DerivedGeology geology(*grid->c_grid(), *new_props, eclipseState, grav);
 
     SimulatorReport fullReport;
     for (size_t reportStepIdx = 0; reportStepIdx < timeMap->numTimesteps(); ++reportStepIdx) {

--- a/examples/sim_fibo_ad_cp.cpp
+++ b/examples/sim_fibo_ad_cp.cpp
@@ -236,7 +236,7 @@ try
     // initialize variables
     simtimer.init(timeMap);
 
-    Opm::DerivedGeology geology(*grid, *new_props, eclipseState);
+    Opm::DerivedGeology geology(*grid, *new_props, eclipseState, grav);
 
     SimulatorReport fullReport;
     for (size_t reportStepIdx = 0; reportStepIdx < timeMap->numTimesteps(); ++reportStepIdx) {

